packages/mwp-app-render/src/components/uxcapture/UXCaptureEventMark.jsx

```diff
@@ -0,0 +1,22 @@
+// @flow
+import * as React from 'react';
+
+type Props = {
+	browserEvent: string,
+	mark: string,
+	children: React$Element<*>,
+};
+
+const generateUXCaptureJS = (mark: string) =>
+	`if(window.UX) { UX.mark('${mark}'); }`;
+
+/**
+ * takes only a *single* child element
+ * and injects event based UX.mark() call into the child
+ *
+ * @see example https://github.com/meetup/ux-capture#image-elements
+ */
+export default ({ browserEvent, mark, children }: Props) =>
+	React.cloneElement(children, {
+		[browserEvent]: generateUXCaptureJS(mark),
```

Comment:
this will also override any existing `[browserEvent]` props in the component, so I think we probably should be a bit more careful about 'merging' with an existing property if there is one, or providing a wrapper.

Reply:
One thing I was wondering is whether we could wrap the child in a `<div>` and set the `onLoad` event to trigger on _that_ rather than the (presumably `<img>`) child - I'm not sure, but I think it's possible that a wrapper's `onLoad` will not fire until all children have loaded, so

```html
<div onload="...">
  <img ...>
</div>
```

might provide the same timings as

```html
<img onload="..." />
```

Reply:
I don't know if we should rely on `onLoad` event even works on `<div>`s

Reply:
I'd rather avoid nesting extra layers just for this.

Reply:
it would be relatively easy to test, and if it works, it would make the interface much more composable at the cost of an extra level of HTML indentation. The major benefit of that approach is that it keeps the implementation 'pure' - the child component is completely unchanged, which is better for predictability (e.g. no concerns about interacting with/overriding an existing `onLoad` prop.

IMO and extra level of HTML indentation is a pretty low cost to pay for predictability

Reply:
It looks like div onload just doesn't work so it has to be on `<img>` HTML tag.

If test-ability is important, maybe custom `<Image>` JSX tag is a better approach?

Reply:
Yeah I like the idea of a dedicated `<Image>` component - we've talked about setting up something like that to encapsulate some more responsive-friendly behavior as well. If we don't want to get into that kind of structure in this PR and save it for a followup, that's fine with me.